### PR TITLE
Use `-Z build-std` with sanitisers.

### PR DESCRIPTION
**Pull request by ltratt**
This forces Rust to rebuild its standard library with sanitiser support built in. I can't observe an effect with address sanitiser, but with thread sanitiser this heavily reduces false positives.

### CODE REVIEW

**Comment by vext01** [comment]
Let's give this a shot.

We should be on the lookout for weirdness. I've just read the docs and this feature has a distinct pre-alpha-quality whiff about it.

bors r+

**Comment by ltratt** [comment]
I *think* the "pre-alphaness" might just be because the docs are outdated. So far it seems pretty solid. Of course I might soon be proven wrong...

**Comment by ltratt** [comment]
@vext01 OK to squash?

File: ykrt/src/mt.rs [hunk 1]
```diff
@@ -1174,11 +1174,12 @@ mod tests {
         let loc = Arc::new(Location::new());
         b.iter(|| {
             let mut thrs = vec![];
-            for _ in 0..4 {
```
[the remark this answers is not in the document]
Comment (ltratt) [reply ykrt/src/mt.rs]: Sausages! Good spot. Reverted in 66b870f.

**Comment by vext01** [comment]
Squash away

**Comment by ltratt** [comment]
Squashed.

**Comment by ltratt** [comment]
It's our old friend, https://github.com/ykjit/yk/issues/874!

**Comment by vext01** [comment]
I'm working on it. Let's retry this for now.

bors r+

**Pull request merged**